File: lab2/lab_2_B_a.py
with open("code.txt", "r") as f:
    content = f.read()
    characters = list(content)

vowels = "aeiou"
consonants = "bcdfghjklmnpqrstvwxyz"

vowel_count = 0
consonant_count = 0

for character in characters:
    if character in vowels:
        vowel_count += 1
    elif character in consonants:
        consonant_count += 1


print(f"Vowel Count: {vowel_count}")
print(f"Consonant Count: {consonant_count}")

File: lab2/lab_2_B_a.c
/* Definitions */
%{
	#include <stdio.h>
	int vowel_count = 0;
    int consonant_count = 0;
    extern FILE *yyin, *yout;
%}

/* Rules */
/* if character belongs to vowel, increment vowel count by 1 */
/* if character belongs to consonant (does not belong to vowel), increment consonant count by 1 */

%%
[ \t\n]+    ;
[aeiouAEIOU]+    {vowel_count++;}
[^aeiouAEIOU]     {consonant_count++;}
%%

/* User subroutines */
int yywrap(){}

int main() {
    yyin = fopen("code.txt", "r");
    yylex();
	printf("Number of vowels = %d\n", vowel_count);
	printf("Number of consonants = %d\n", consonant_count);
    return 0;
}
